rfcs/text/0000-dss-deletion-daemon.md

```diff
@@ -0,0 +1,620 @@
+     "email": "<requester's email>"
+   }
+   ```
+   
+   A versioned bundle tombstone will cause `GET /bundle/{uuid}` to return a 409 for that bundle version. If the tombstoned
+   bundle is the latest version then a request for `GET /bundle/{uuid}` without a version will
+   also returns a 409. If bundle tombstone does not specify a bundle version then all request for `GET /bundle/{uuid}` will
+   return 409 even if version is specified in the request. The same applies to `HEAD /bundle/{uuid}` requests, both versioned and
+   unversioned. The same applied for `GET /files/{uuid}` and `HEAD /files/{uuid}`. Bundles that have an 
+   associated tombstone are inaccessible through the API and are considered **logically deleted**. Files with an 
+   associated tombstone or not logically delete. User with a direct URL to the file can still access the data.
+   
+   A notification is sent out to subscribers of datastore deletions when a new bundle tombstone is created, so they can 
+   update their index appropriately. If the the bundle was logically deleted and later physically deleted a second
+   identical notification will be sent to subscribers. The subscribers must be able to handle these notifications 
+   idempotently.
+   
+   A **physical deletion** request of a bundles results in the bundle associated files being added to 
+   the deletion queue for regularly schedule **Physical Deletion**. A **physical deletion** request of a file result in 
+   that file being added to the to deletion queue
+
+5) On a regular schedule the deletion queue is processed by code running within a secure boundary. An administrator 
+   may manually invoke the deletion process or wait for it run at it's scheduled time once daily.
+   The deletion daemon is responsible for performing **physical deletes** of data from the deletion queue. 
+   A physical delete makes all files associated with a bundle or bundle version inaccessible. This will cause direct
+   urls to a deleted file to fail.
+   
+   A log entry is produced when ever the deletion daemon is run. It contains information on what bundles and files
+   have been logically deleted, the reason, the administrator who executed the deletion, and info about the deletion markers. 
+   The same information is stored in the **Deletion Table**. The deletion table tracks all bundles and files currently
+   pending permanent deletion. The Deletion Table contains enough information to reverse Physical Deletes before 
+   the data is Permanently deleted.
+   
+   A daily digest of bundles and files that are scheduled for deletion are sent to DCP administrator 24 hours prior to the data
+   being permanently deleted.
+
+6) Data that has been physical deleted remain in the bucket for 7 days before being permanently deleted from the bucket.
+   This grace period allows the deletion to be reversed within that time period. After the grace period the data will 
+   be permanently deleted.
+ 
+### Tombstones
```

Comment:
Tombstones are really an implementation detail of logical deletion within DSS. I'm not sure that external users really need to know about them.